src/makefile

```diff
@@ -34,22 +34,6 @@ endif
 .SUFFIXES:
 .PHONY: default userspace modules clean modclean depclean install python pythonclean cscope cscopeclean
 
-# A "trivial build" is one which should not include dependency information
-# either because it should be usable before dependency information can be
-# generated or when it is invalid (clean, docclean) or when running as root
-# when the user must guarantee in advance that everything is built
-# (setuid, install)
-ifeq ($(MAKECMDGOALS),)
```

Comment:
Can you please let me know the motivation of removing TRIVIAL_BUILD? It causes "make clean" to have to do a substantial amount of work. Additionally, we have found in the past that if (e.g.,) "sudo make setuid" does things like create dependency files and their containing directories, users end up with problems removing the files later with "make clean". Users are not always aware of whether invoking "make" will need to do these steps, since a point of make is to abstract away details like these.

Reply:
I see that the initial comment explains this:
> Removes everything related to TRIVIAL_BUILD as in a standard automake project it is also not possible to invoke any make targets prior to generating configure/Makefile files.
However, as the automake'd Makefile is never actually used, does it require breaking this feature?